### ISSUE TITLE
ENH: representative_point as an attribute

### ISSUE THREAD

**Issue by martinfleis** [geopandas/geopandas#1616]
We have `centroid`, `exterior` or `envelope` as attributes, but `representative_point()` as a method. Would be good to have an alias as an attribute, to have them all consistent. There's no potential for any arguments anyway afaik.

**Comment by jorisvandenbossche** [comment]
Yeah, this is not very consistent. It was added to shapely here (https://github.com/Toblerity/Shapely/commit/cead5986637fc7b007a7624a17a045e881c4c007), but don't directly see an indication why a method was used here.

Changing it to an attribute is a breaking change, but I am also not sure it is worth adding another alias for it? 

(BTW, in general I think it is also "misleading" that all those others are attributes and not methods, as they are not all cheap properties, but sometimes the result of an expensive calculation. But yeah, that's not something we can easily change .. ;-))

**Comment by martinfleis** [comment]
> did you have a certain name in mind?

Initially not, but thinking about it I realised that shapely is the only case I can think of where this function is called `representative point`.

`point_on_surface` (PyGEOS)
`ST_PointOnSurface` (PostGIS)
`st_point_on_surface` (sf)
`Point on surface` (QGIS).

So it may be worth adding `point_on_surface` attribute for the sake of consistency 1) internally with other attributes (although they're not cheap...), and 2) externally with other geospatial tools.

**Comment by jorisvandenbossche** [comment]
So the original commit I mentioned above that renamed it from a `point_on_surface` property to `representative_point()` method says *"rename to representative_point following OGC 06-103r3, and since it applies to objects other than surfaces."*

Looking at https://www.ogc.org/standards/sfa, it indeed mentions "representativePoint", but as method of the GM_Object (where the equivalent method is PointOnSurface), which I am not really what that is. And eg https://www.ogc.org/standards/sfs only mentions PointOnSurface (which is also shown by the fact that GEOS, Postgis etc all call it like that).

So adding a `point_on_surface` property indeed seems fine, certainly since that aligns with how similar libraries call it. 
But, ideally that would also be done then in Shapely, though, to keep shapely and geopandas consistent. cc @sgillies

**Comment by sgillies** [comment]
I made representative_point is a method because the algorithm yields *a* point guaranteed to intersect a surface. If it returned a slightly different value every time, it would still conform to the spec. Centroid, exterior, envelope, area, all these have a 1:1 relationship with a geometric object and attributes are a good fit.
